test/java/seedu/addressbook/logic/LogicTest.java

```diff
@@ -90,7 +90,10 @@ private void assertCommandBehavior(String inputCommand,
         //Confirm the state of data is as expected
         assertEquals(expectedAddressBook, addressBook);
         assertEquals(lastShownList, logic.getLastShownList());
-        assertEquals(addressBook, saveFile.load());
+        if (inputCommand.equals("add") || inputCommand.equals("clear") ||
```

Comment:
You have found the right place to update the test. Now think about how to elegantly do it. Since you are checking the command types, consider what class helps you do this in the original code, which returns you a Command object. After that, you can test the isMutating() method directly. Lastly, think about how to test the else branch, i.e. when the method is not mutating.